fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexerIndexFromIndexTest.java

```diff
@@ -0,0 +1,365 @@
+/*
+ * OnlineIndexerIndexFromIndexTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2020 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.record.TestRecords1Proto;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.IndexOptions;
+import com.apple.foundationdb.record.metadata.IndexTypes;
+import com.apple.foundationdb.record.metadata.expressions.EmptyKeyExpression;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+
+import static com.apple.foundationdb.record.metadata.Key.Expressions.field;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for building indexes from other indexes with {@link OnlineIndexer}.
+ */
+
+public class OnlineIndexerIndexFromIndexTest extends OnlineIndexerTest {
```

Comment:
It looks like this is missing tests on:

1. Multi-type indexes
1. The source index is on the wrong type(s)
1. The source index is not a `VALUE` index
1. There are concurrent builds
1. An index build from an index takes over from an index build without an index and vice versa
1. The meta-data change midway through a build (might be somewhat hard to simulate)

It would also be nice if the tests validated that the index looked correct (perhaps by running an index validation on the resulting index). I'm not sure if the tests in the other files do that or not, though perhaps they should as well.

If you can think of other test cases, @nblintao, it would be good to enumerate them, too.

Reply:
Did these tests get added?

Reply:
Only the third. Still struggling with the others.